tools/c7n_azure/c7n_azure/resources/network_security_group.py

```diff
@@ -171,15 +173,21 @@ def get_port_range(self, range_str):
         {} - Lower bound of port range (inclusive)
         {} - Upper bound of port range (inclusive)
         {} - TCP/UDP protocol
-    """.format(PORTS, EXCEPT_PORTS, FROM_PORT, TO_PORT, IP_PROTOCOL)
+        {} - Allow/Deny access
+    """.format(PORTS, EXCEPT_PORTS, FROM_PORT, TO_PORT, IP_PROTOCOL, ACCESS)
 
     def is_match(self, security_rule):
         if self.direction_key != security_rule['properties']['direction']:
             return False
         ranges_match = self.is_ranges_match(security_rule)
         protocol_match = (self.ip_protocol is None) or \
                          (self.ip_protocol == security_rule['properties']['protocol'])
-        return self.match_op([ranges_match, protocol_match])
+
+        if self.access is not None:
+            access_match = self.access == security_rule['properties']['access']
```

Comment:
Suggestion - we found a bug recently from case sensitive string matching. To be safe maybe add a .lower() to both in case our azure folks decide to switch casing up on us.

Reply:
Oh man just noticed we have string matchings everywhere here. Hmmm...